docs/Coding-Guidelines.md

```diff
@@ -1,34 +1,49 @@
-#### General
+# General
 
-Project Reunion prefers using industry-standard coding styles, guidelines, and patterns for any languages used in implementation or testing.
+Project Reunion prefers using industry-standard coding styles, guidelines, and patterns for any
+languages used in implementation or testing.
 
-#### C++
+## Languages
 
-Project Reunion implementations prefer C++ and C++/WinRT implementations.   New code must adhere to the **cppcoreguidelines** at
-https://github.com/isocpp/CppCoreGuidelines and be /W4 clean (for Visual C++.)
+This list is non-exhaustive; new guidelines for languages will be added over time.
 
-Prefer existing std:: and gsl:: types, but use **wil** (https://github.com/Microsoft/wil) types for Windows-specific helpers and lifecycle management
-rather than creating your own.
+### C++
 
-##### Catching Exceptions and HRESULT
+**DO** implement types using C++ and [C++/WinRT](https://github.com/microsoft/cppwinrt)
 
-Don't just catch `winrt::hresult_error` or other variations as that doesn't catch `std::bad_alloc`.
+**DO** follow the [CppCoreGudelines](https://github.com/isocpp/CppCoreGuidelines) for all new code.
 
-Use the following snippet to catch exceptions and retrieve their HRESULT:
+**DO** enable `/W4 /Wx` for all new code.
+
+**CONSIDER** using the [Windows Implementation Library](https://github.com/Microsoft/wil) for
+Windows-specific helpers rather than creating your own.
+
+**DO** use 4-space indentation instead of tab characters.
```

Comment:
there is a lot more to say about code formatting. this is pedantic, but needs to be specified. can we do this in terms of a .clang-format file?

Reply:
Sure thing.  Please propose a .clang-format file as a normal pull request. Do note the "use existing style" though, so we might _avoid_ reformatting swathes of code that were lifted from an internal source so we can more easily port changes.